frontend/src/components/student/studentStatusButton.tsx

```diff
@@ -0,0 +1,41 @@
+import { Button } from "react-bootstrap";
+import { Status } from "../../api/entities/StudentEntity";
+import { useRouter } from "next/router";
+import { useEffect, useState } from "react";
+
+/**
+ * The StudentStatusButton filters the list of students by the passed student status
+ * @param props a status and a colour for the button
+ */
+export function StudentStatusButton(props: { status: Status; colour: string }) {
+    const [clicked, setClicked] = useState(false);
+    const router = useRouter();
+
+    useEffect(() => {
+        setClicked(router.query?.status === props.status);
+    }, [router.query.status]);
```

Comment:
One has null safety and one does not? No idea if your props need to be in the deps array?
```suggestion
   const status = router.query.status;
    useEffect(() => {
        setClicked(status === props.status);
    }, [status]);
```